angular/src/components/login.component.ts

```diff
@@ -132,7 +132,12 @@ export class LoginComponent extends CaptchaProtectedComponent implements OnInit
 
     togglePassword() {
         this.showPassword = !this.showPassword;
-        document.getElementById('masterPassword').focus();
+        const masterPassword: HTMLInputElement = <HTMLInputElement>document.getElementById('masterPassword');
```

Comment:
Code style: no need to specify the type of the `const`, it'll figure it out from the value being assigned.